sklearn/neural_network/tests/test_rbm.py

```diff
@@ -189,3 +190,18 @@ def test_sparse_and_verbose():
                         r" time = (\d|\.)+s", s)
     finally:
         sys.stdout = old_stdout
+
+
+@pytest.mark.parametrize("dtype_in, dtype_out", [
+    (np.float32, np.float32),
+    (np.float64, np.float64),
+    (np.int, np.float64)])
+def test_transformer_dtypes_casting(dtype_in, dtype_out):
+    X = Xdigits[:100].astype(dtype_in)
+    rbm = BernoulliRBM(n_components=16, batch_size=5, n_iter=5,
+                       random_state=42)
+    Xt = rbm.fit_transform(X)
+
+    # dtype_in and dtype_out consistent
+    assert Xt.dtype == dtype_out, ('transform dtype: {} - original dtype: {}'
+                                   .format(Xt.dtype, X.dtype))
```

Comment:
While https://github.com/scikit-learn/scikit-learn/pull/16290 is not merged, we also should add a check that the results of `fit_transform` are close enough with float32 and float64 input.

Reply:
We also should check that the attributes of the estimator are close on float32 and float64 inputs, which cannot be done in the common test.

Reply:
Just added a dedicated test for that.